### PR TITLE
chore(ci): add ci workflow for spelunker-web

### PR DESCRIPTION
**Pull request by fallenoak**
This PR introduces a CI workflow for `spelunker-web`.

### CODE REVIEW

File: .github/workflows/web-ci.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        path: ~/.npm
+    - name: Install Node modules
+      run: npm install
+      working-directory: ./packages/spelunker-web
+    - name: Lint
+      run: npm run lint
+      working-directory: ./packages/spelunker-web
+    - name: Build
+      run: npm run build
+      working-directory: ./packages/spelunker-web
```
Comment (timkurvers) [inline .github/workflows/web-ci.yml:45]: If all of the commands are in this working directory, I suppose we could use the following in the job definition:

```yaml
defaults:
  run:
    working-directory: ./packages/spelunker-web
```

That could potentially influence the `.npm`-thing above though, so unsure.

**Review by timkurvers — APPROVED** [review]
🥳

File: .github/workflows/web-ci.yml (hunk 1 above)
Comment (fallenoak) [reply .github/workflows/web-ci.yml:45]: Oh interesting. I'll see how it works in a subsequent PR. I'm looking at using a conditional paths action in a pre-run job to make the required checks work better, too.

Comment (timkurvers) [reply .github/workflows/web-ci.yml:45]: 👍

**Pull request merged**